README.md

```
@@ -1,11 +1,61 @@
+cargo build --workspace --all-features
+
+# Run tests
+cargo nextest run --workspace --all-features --all-targets
+
+# Run the interactive CLI
+cargo run --bin kv-engine-cli -- --path /tmp/lsm.db --compaction leveled
+```
+
+## Architecture
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)